app/controllers/GuidanceController.scala

```diff
@@ -133,34 +132,43 @@ class GuidanceController @Inject() (
         case err @ Left(_) => err
       }
 
-    val sessionId: String = hc.sessionId.fold(java.util.UUID.randomUUID.toString)(_.value)
-    logger.info(s"Starting approval direct view with sessionId = $sessionId")
-    retreiveCacheAndRedirectToProcess(processId, sessionId, retrieveCacheAndRedirect(s"/$url"))
+    logger.info(s"Starting approval direct page view journey")
+    retreiveCacheAndRedirectToView(processId, retrieveCacheAndRedirect(s"/$url"))
   }
 
-  private def withSession[T](block: String => Future[RequestOutcome[T]])(implicit request: Request[_]): Future[RequestOutcome[T]] =
+  private def withExistingSession[T](block: String => Future[RequestOutcome[T]])(implicit request: Request[_]): Future[RequestOutcome[T]] =
     hc.sessionId.fold {
       logger.error(s"Session Id missing from request when required")
       Future.successful(Left(BadRequestError): RequestOutcome[T])
     } { sessionId =>
-      logger.info(s"withSession, EG_SESSIONID ${request.session.get("EG_SESSION")}, sessionId $sessionId")
+      logger.info(s"Found existing sessionId = $sessionId")
       block(sessionId.value)
     }
 
-  private def retreiveCacheAndRedirectToProcess(id: String, sessionId: String, processStartUrl: (String, String) => Future[RequestOutcome[String]])(
+  private def retreiveCacheAndRedirectToView(id: String, retreiveAndCache: (String, String) => Future[RequestOutcome[String]])(
       implicit request: Request[_]
-  ): Future[Result] =
-    processStartUrl(id, sessionId).map {
+  ): Future[Result] = {
+    val (sessionId, egNewSessionId) = existingOrNewSessionId()
+    logger.info(s"Calling Retreive and cache service for process $id using sessionId = $sessionId, EG = ${egNewSessionId}")
+    retreiveAndCache(id, sessionId).map {
       case Right(url) =>
-        logger.warn(s"Redirecting to begin viewing process $id at ${routes.GuidanceController.getPage(url.drop(1)).toString} using sessionId $sessionId")
-        Redirect(routes.GuidanceController.getPage(url.drop(1))).withSession((SessionKeys.sessionId -> sessionId), ("EG_SESSION" -> sessionId))
+        val target = routes.GuidanceController.getPage(url.drop(1)).toString
+        logger.warn(s"Redirecting to begin viewing process $id at ${target.toString} using sessionId $sessionId, EG_NEW_SESSIONID = $egNewSessionId")
+        egNewSessionId.fold(Redirect(target))(newId => Redirect(target).addingToSession(("EG_NEW_SESSIONID" -> newId)))
       case Left(NotFoundError) =>
         logger.warn(s"Unable to find process $id and render using sessionId $sessionId")
         NotFound(errorHandler.notFoundTemplate)
       case Left(err) =>
         logger.error(s"Error $err returned from Guidance service when trying to access process $id using sessionId $sessionId")
         InternalServerError(errorHandler.internalServerErrorTemplate)
     }
+  }
+
+  private def existingOrNewSessionId()(implicit request: Request[_]): (String, Option[String]) = 
```

Comment:
IntelliJ is reporting trailing white space at the end of the line.

Reply:
will remove